src/vm/ilmarshalers.cpp

```diff
@@ -2693,10 +2578,9 @@ MarshalerOverrideStatus ILSafeHandleMarshaler::ArgumentOverride(NDirectStubLinke
 
     if (fManagedToNative)
     {
+        pslIL->SetStubTargetArgType(ELEMENT_TYPE_I);
```

Comment:
I don't understand why this was pulled out of the `if`.

Reply:
Oh, is this because we are adding IL in both scenarios now?

Reply:
Yep. We're passing a `nativeint` down to native in both cases so I moved it out of the if.